File: SMTH/reverse.java
import java.util.Arrays;

public class reverse{
    public static int[] reverseit(int[] arr){
        int[] rev = new int[arr.length];
        for(int i = 0; i<arr.length;i++)
        {
            rev[i] = arr[arr.length-i-1];
        }
        return rev;
    }
    public static void main(String[] args){
        int[] ar = {4,5,6,3,10,1};
        int[] revarray = reverseit(ar);
        System.out.println(Arrays.toString(revarray));
    }
}

File: Array/reverse.cpp
#include<iostream>

using namespace std;

void rev(int arr[], int n){
    int s = 0;
    int e = n-1;
    while(s<e){
        swap(arr[s],arr[e]);
        s++;
        e--;
    }
}

int main(){
    int arr[] = {3,4,5,6,7,8};
    int size = sizeof(arr)/sizeof(arr[0]);
    rev(arr, size);
    for(auto x : arr){
        cout << x << " ";
    }
}